sql/src/main/java/io/crate/metadata/sys/SysTableDefinitions.java

```diff
@@ -85,10 +83,15 @@ public SysTableDefinitions(JobsLogs jobsLogs,
             SysRepositoriesTableInfo.expressions()
         ));
         tableDefinitions.put(SysSnapshotsTableInfo.IDENT, new StaticTableDefinition<>(
-            snapshotSupplier(sysSnapshots),
```

Comment:
I'd be in favor of keeping a simple wrapper function instead of relying on inheritance. The `snapshotSupplier` function could be made generic, just have to change the argument to `Supplier<Iterable<T>>`

The simple wrapper function imho leads to better decoupling, because there is no reason for the SysSnapshots/SysAllocations to return the items as future, the data is available without doing any I/O. By exposing it as future we're making it specific to this particular use-case